### DISCUSSION TITLE
How to extend from turbo?

### DISCUSSION THREAD

**Discussion by thany** [typescript-eslint/typescript-eslint#10400] [Technical Discussions]
### Body

Please consider this eslint config:

```
// @ts-check
import turbo from 'eslint-config-turbo/flat';
import tseslint from 'typescript-eslint';

export default tseslint.config(
  ...turbo,
  {
    // ...other irrelevant config
  }
);
```

It reports this error:

```
Argument of type '{ plugins: { turbo: typeof plugin; }; rules: { "turbo/no-undeclared-env-vars": string; }; }' is not assignable to parameter of type 'InfiniteDepthConfigWithExtends'.
  Types of property 'rules' are incompatible.
    Type '{ "turbo/no-undeclared-env-vars": string; }' is not assignable to type 'Partial<Record<string, RuleEntry>>'.
      Property '"turbo/no-undeclared-env-vars"' is incompatible with index signature.
        Type 'string' is not assignable to type 'RuleEntry | undefined'.ts(2345)
```

There's not much I can do about this. The `string` it complains about, happens to be `'error'`, so I don't understand:

1. Why isn't it imported more strictly so that it matches the expected type?
2. What am I supposed to do about this?

So I can't just do `turbo as ConfigWithExtends` or whatever, because ESLint requires the config to be a javascript file. So in fact there's literally nothing I can add to make typescript understand what it's importing.

And also, I'm not the author of `eslint-config-turbo` so I can't just add typings to it. But moreover, you can't expect every author of every eslint config package to provide their typings.

Afaict, this is placing me between a rock and a hard place. What do I do?

### Additional Info

Any additional info...

### Before you submit your discussion, please confirm the following. If any of these required steps are not taken, we may not be able to review your RFC. Help us to help you!

- [X] I have [read the discussions guidelines](https://typescript-eslint.io/contributing/discussions) and per those guidelines, this fits a discussion category, _not_ a help request or standard issue.
- [X] I have [searched for related discussions](https://github.com/typescript-eslint/typescript-eslint/discussions) and [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

**Comment by bradzacher** [comment]
As per our discussion guidelines (which you ticked as having read): https://typescript-eslint.io/contributing/discussions/
We do not use discussions as a support forum. Please read the guidelines for alternative.
